### PR TITLE
[UPDATE] Updated plaidkey detector results, and added uniqueness check

### PR DESCRIPTION
**Pull request by nabeelalam**
### Description:
Updated the `plaidkey` detector.
Makes it so that only unique keys and IDs will be processed.
Also adds the `s1` object into the `results` when the `verify` flag is `false` in order to get required output in pattern test.

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/welcome/install/#local-installation))?

### CODE REVIEW

File: pkg/detectors/plaidkey/plaidkey.go [hunk 1]
```diff
@@ -37,20 +38,22 @@ func (s Scanner) Keywords() []string {
...
+		uniqueKeys[foundKey[1]] = struct{}{}
+	}
 
-		for _, idMatch := range idMatches {
-			if len(idMatch) != 2 {
-				continue
-			}
-			idresMatch := strings.TrimSpace(idMatch[1])
+	for _, foundId := range idPat.FindAllStringSubmatch(dataStr, -1) {
+		uniqueIds[foundId[1]] = struct{}{}
```
Comment (rgmz) [inline pkg/detectors/plaidkey/plaidkey.go:49]: Eliminates stuff like `PLAID_CLIENT_ID=XXXXXXXXXXXXXXXXXXXXXXXX`.
```suggestion
		id := foundId[1]
		if detectors.StringShannonEntropy(id) < 3 {
			continue
		}

		uniqueIds[id] = struct{}{}
```

File: pkg/detectors/plaidkey/plaidkey.go [hunk 2]
```diff
@@ -37,20 +38,22 @@ func (s Scanner) Keywords() []string {
...
+	// find all the matching keys and ids in the data and make a unique maps for both.
+	uniqueKeys, uniqueIds := make(map[string]struct{}), make(map[string]struct{})
 
-	for _, match := range matches {
-		if len(match) != 2 {
-			continue
-		}
-		resMatch := strings.TrimSpace(match[1])
+	for _, foundKey := range keyPat.FindAllStringSubmatch(dataStr, -1) {
+		uniqueKeys[foundKey[1]] = struct{}{}
```
Comment (rgmz) [inline pkg/detectors/plaidkey/plaidkey.go:45]: Eliminates stuff like `PLAID_SECRET=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX`.
```suggestion
		key := foundKey[1]
		if detectors.StringShannonEntropy(key) < 3 {
			continue
		}

		uniqueKeys[key] = struct{}{}
```

File: pkg/detectors/plaidkey/plaidkey.go
Comment (rgmz) [inline pkg/detectors/plaidkey/plaidkey.go:1]: The detector should be updated to handle failures and check for a specific status code. Anything that uses `if err == nil` or `if res.StatusCode >= 200 && res.StatusCode < 300` is defective IMO (#3658).

See the implementation [here](https://github.com/trufflesecurity/trufflehog/blob/31b4dc2fb7633124245f30b3eb8e1b35e031ee39/pkg/detectors/alchemy/alchemy.go#L51-L59).

File: pkg/detectors/plaidkey/plaidkey.go (hunk 1 above)
Comment (nabeelalam) [reply pkg/detectors/plaidkey/plaidkey.go]: Thanks @rgmz! I didn't think of this, but any detectors I may work on later I'll definitely add this!

**Review by kashifkhan0771 — APPROVED** [review]

**Pull request merged**